### PR TITLE
#359 - Added the create, clone and delete scope functions

### PR DESCRIPTION
**Pull request by raduiordache**
Added also tests

### CODE REVIEW

**Comment by raduiordache** [comment]
Hallo @jberends,
Kannst du meine Kode prufen?
Meine Deutsch ist nicht so gut aber ich denke dass du kannst mich verstehen.

Danke schon

File: pykechain/client.py [hunk 1]
```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
...
+        :param start_date: (optional) start date of the scope
+        :type start_date: datetime.datetime or None
+        :param due_date: (optional) due date of the scope
+        :type due_date: datetime.datetime or None
+        :param team: Team object
+        :type team :class: `models: Team`
+        :return: the created :class:`models.Scope`
+        :raises APIError: In case of failure of the creation of new Scope
+        """
+        assert isinstance(name, (str, text_type))
```
Comment (jberends) [inline pykechain/client.py]: assert is not robust enough when you [run python in optimised mode](https://github.com/emre/notes/blob/master/python/when-to-use-assert.md).

> if you run Python with the -O or -OO optimisation flags, and consequently there is no guarantee that assert statements will actually be run.



```suggestion
        if not isinstance(name, (str, text_type)):
            raise IllegalArgumentError(....)        
```

File: pykechain/client.py [hunk 2]
```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
 
         return service
 
+    def create_scope(self, name, status=None, description=None, tags=None, start_date=None, due_date=None,
+                     team=None):
```
Comment (jberends) [inline pykechain/client.py]: add `**kwargs` to the create_scope as well to ensure forwards compatibility

File: pykechain/client.py [hunk 3]
```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
...
+        data = {
+            'name': name,
+            'status': status,
+            'text': description,
+            'tags': tags,
+            'start_date': start_date,
+            'due_date': due_date,
+            'team_id': team.id if team else None,
+        }
+
```
Comment (jberends) [inline pykechain/client.py]: incorporate kwargs

```suggestion
        data.update(kwargs)
```

File: pykechain/client.py [hunk 4]
```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
...
+        :raises APIError: in case of failure in the deletion of the scope
+        """
+        assert isinstance(scope, Scope), 'Scope "{}" is not a scope!'.format(scope.name)
+
+        response = self._request('DELETE', self._build_url('scope', scope_id=str(scope.id)))
+
+        if response.status_code != requests.codes.no_content:  # pragma: no cover
+            raise APIError("Could not delete scope, {}: {}".format(str(response), response.content))
+
+    def clone_scope(self, scope, name=None, status=None, description=None, tags=None, start_date=None, due_date=None,
```
Comment (jberends) [inline pykechain/client.py]: did we check in the PIM2 branch if there is already this code?

incorporate `**kwargs`

File: pykechain/models/scope.py [hunk 5]
```diff
@@ -398,3 +398,24 @@ def edit(self, name=None, description=None, start_date=None, due_date=None, stat
...
+
+    def clone(self, *args, **kwargs):
+        """
+        Clone current scope.
+
+        See :class:`pykechain.Client.clone_scope` for available parameters.
+
+        .. versionadded:: 2.6.0
+        """
+        return self._client.clone_scope(*args, scope=self, **kwargs)
```
Comment (jberends) [inline pykechain/models/scope.py]: 👍

**Pull request merged**